src/processes/news-process.js

```diff
@@ -1,213 +1,250 @@
+    let contentPublishedAt = null;
+
+    /*
           @description crawls attribut opengraph url domain
         */
-        const $ = cheerio.load(crawls.data);
-        const site_name = $('meta[property="og:site_name"]').attr('content') || "";
-        const title = $('meta[property="og:title"]').attr('content') || "";
-        const image = $('meta[property="og:image"]').attr('content') || "";
-        let description = $('meta[property="og:description"]').attr('content') || "";
-        // if (description === "") {
-        //   description = $('meta[name="description"]').attr('content') || "";
-        // }
-        const news_url = $('meta[property="og:url"]').attr('content') || "";
-        const url_compact = $('meta[property="og:url"]').attr('content') || "";
-        const keyword = $('meta[name="keywords"]').attr('content') || "";
-        const author = $('meta[name="author"]').attr('content') || "";
-
-        /*
+    const $ = cheerio.load(crawls.data);
+
+    const site_name = $('meta[property="og:site_name"]').attr('content') || '';
+    const title = $('meta[property="og:title"]').attr('content') || '';
+    const image = $('meta[property="og:image"]').attr('content') || '';
+    const description = $('meta[property="og:description"]').attr('content') || '';
+    // if (description === "") {
+    //   description = $('meta[name="description"]').attr('content') || "";
+    // }
+    const news_url = $('meta[property="og:url"]').attr('content') || '';
+    const url_compact = $('meta[property="og:url"]').attr('content') || '';
+    const keyword = $('meta[name="keywords"]').attr('content') || '';
+    const author = $('meta[name="author"]').attr('content') || '';
+    try {
+      const ldJson = $("script[type='application/ld+json']")[0]?.children?.at(0)?.data;
+      const ldParsedJson = JSON.parse(ldJson);
+      contentPublishedAt = ldParsedJson?.datePublished;
+    } catch (e) {
+      console.error(e);
+    }
+
+    /*
           @description define data for post to getstream and newslink table
         */
-        let data = {
-            domain_page_id, title, site_name, image, description, news_url, keyword, author, url_compact
-        };
-
-
-        /*
+    const data = {
+      domain_page_id,
+      title,
+      site_name,
+      image,
+      description,
+      news_url,
+      keyword,
+      author,
+      url_compact,
+      contentPublishedAt
+    };
+
+    /*
           @description save news link and post to getstream
         */
-        const result = await saveNewsLink(data, name, info, job.data, logo, created_domain);
-        done(null, result);
-    } catch (error) {
-        console.error('kesalahan', error);
-        done(null, error);
-    }
-}
+    const result = await saveNewsLink(data, name, info, job.data, logo, created_domain);
+    done(null, result);
+  } catch (error) {
+    console.error('kesalahan', error);
+    done(null, error);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In case of an error, the job is marked as done and the error is passed as the result. This could be misleading as it might seem that the job has completed successfully when in fact an error occurred. It would be better to pass the error to the `done` callback as the first argument, which will mark the job as failed and store the error.
```diff
-     console.error('kesalahan', error);
-     done(null, error);
+     done(error);
```

Overall, the code seems to lack proper error handling. There are several places where errors are caught and logged, but then execution continues as if nothing happened. This can lead to unexpected behavior. It's generally a good practice to either handle the error or fail fast.


<!-- This is an auto-generated comment by OSS CodeRabbit -->